tensorboard/backend/event_processing/db_import_multiplexer_test.py

```diff
@@ -0,0 +1,156 @@
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import os.path
+
+from tensorboard.backend import application
+from tensorboard.backend.event_processing import db_import_multiplexer
+from tensorboard.compat.proto import event_pb2
+from tensorboard.compat.proto import summary_pb2
+from tensorboard.compat.proto import tensor_pb2
+from tensorboard.compat.proto import types_pb2
+import tensorflow as tf
+
+
+def _AddEvents(path):
+  with tf.summary.FileWriter(path) as writer:
+    event = event_pb2.Event(
+        summary=summary_pb2.Summary(
+            value=[
+                summary_pb2.Summary.Value(
+                    tensor=tensor_pb2.TensorProto(
+                        dtype=types_pb2.DT_INT32,
+                        int_val=[1],
+                    )
+                )
+            ]
+        )
+    )
+    native_event = tf.Event()
+    native_event.ParseFromString(event.SerializeToString())
+    writer.add_event(native_event)
+
+
+class DbImportMultiplexerTest(tf.test.TestCase):
+
+  def setUp(self):
+    super(DbImportMultiplexerTest, self).setUp()
+
+    db_uri = 'sqlite:%s/db' % (self.get_temp_dir())
+    _, self.db_connection_provider = application.get_database_info(db_uri)
+    self.multiplexer = db_import_multiplexer.DbImportMultiplexer(
+        db_connection_provider=self.db_connection_provider,
+        purge_orphaned_data=False,
+        max_reload_threads=1,
+        use_import_op=False)
+
+  def _getRuns(self):
+    db = self.db_connection_provider()
+    cursor = db.execute('''
+      SELECT
+        Runs.run_name
+      FROM Runs
+    	ORDER BY Runs.run_name
+    ''')
+    return [row[0] for row in cursor]
+
+  def _getExperiments(self):
+    db = self.db_connection_provider()
+    cursor = db.execute('''
+      SELECT
+        Experiments.experiment_name
+      FROM Experiments
+      ORDER BY Experiments.experiment_name
+    ''')
+    return [row[0] for row in cursor]
+
+  def test_init(self):
+    """Tests that DB schema is created when creating DbImportMultiplexer."""
+    # Reading DB before schema initialization raises.
+    self.assertEqual(self._getExperiments(), [])
+    self.assertEqual(self._getRuns(), [])
+
+  def testAddRunsFromDirectory_empty_folder(self):
+    fake_dir = os.path.join(self.get_temp_dir(), 'fake_dir')
+    self.multiplexer.AddRunsFromDirectory(fake_dir)
+    self.assertEqual(self._getExperiments(), [])
+    self.assertEqual(self._getRuns(), [])
+
+  def testAddRunsFromDirectory_flat(self):
+    path = self.get_temp_dir()
+    _AddEvents(path)
+    self.multiplexer.AddRunsFromDirectory(path)
+    self.multiplexer.Reload()
+    # Because we added runs from `path`, there is no folder to infer experiment
+    # and run names from.
+    self.assertEqual(self._getExperiments(), [u'.'])
+    self.assertEqual(self._getRuns(), [u'.'])
+
+  def testAddRunsFromDirectory_single_level(self):
```

Comment:
Can you add a test for the common mixed-level case where there are events at the top level as well as under child subdirs?  E.g. the common "train data in /, eval data in /eval" pattern.

I believe that will result in experiments == [".", "eval"] and runs = [".", "."].